language/move-lang/src/parser/syntax.rs

```diff
@@ -1311,13 +1311,13 @@ fn parse_module<'input>(tokens: &mut Lexer<'input>) -> Result<ModuleDefinition,
     }
 
     let mut structs = vec![];
-    while is_struct_definition(tokens)? {
-        structs.push(parse_struct_definition(tokens)?);
-    }
-
     let mut functions = vec![];
     while tokens.peek() != Tok::RBrace {
-        functions.push(parse_function_decl(tokens, /* allow_native */ true)?);
+        if is_struct_definition(tokens)? {
```

Comment:
I think this is likely to cause issues in the future if there were a lot of shared modifiers? But it's probably fine for now

Reply:
It might become a little more of a hassle, but it shouldn't be that hard, even with more shared modifiers. We can just parse the modifiers into flags and then pass those flags down.